### PR TITLE
tests: sunset tmlibs/process.Process

### PR DESCRIPTION
**Pull request by odeke-em**
Updates https://github.com/tendermint/tmlibs/issues/81

No longer using tmlibs/process.Process as we deemed
it racy and would incur a maintenance cost yet not
used anywhere else but in these tests and not in actual
code.

### CODE REVIEW

File: tests/test_app/main.go [hunk 1]
```diff
@@ -27,8 +30,17 @@ func testCounter() {
 	}
 
 	fmt.Printf("Running %s test with abci=%s\n", abciApp, abciType)
-	appProc := startApp(abciApp)
-	defer appProc.StopProcess(true)
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("abci-cli %s", abciApp))
+	cmd.Stdout = os.Stdout
+	if err := cmd.Start(); err != nil {
+		log.Fatalf("starting %q err: %v", abciApp, err)
```
Comment (melekes) [inline tests/test_app/main.go]: shouldn't we panic?

Comment (odeke-em) [reply tests/test_app/main.go]: Same thing but here with more context.

File: tests/test_app/main.go [hunk 2]
```diff
@@ -27,8 +30,17 @@ func testCounter() {
...
+	}
+	defer cmd.Wait()
+	defer cmd.Process.Kill()
+
+	// TODO: A better to wait for the app to start?
```
Comment (ebuchman) [inline tests/test_app/main.go]: yes, we should be able to use `abci-cli echo` repeatedly until it works!

Comment (odeke-em) [reply tests/test_app/main.go]: Would you like me to incorporate this change in right now? That sleep was a port over from the original code :)

Comment (ebuchman) [reply tests/test_app/main.go]: Yeh I noticed. But I also fixed a bunch of these recently and non-deterministic tests are the worst, so would love if you fixed this!

Comment (odeke-em) [reply tests/test_app/main.go]: Awesome, coming up.

Comment (odeke-em) [reply tests/test_app/main.go]: C'est fini @ebuchman, merci beaucoup!

**Comment by ebuchman** [comment]
@odeke-em see fix for ensureABCIIsUp

**Comment by odeke-em** [comment]
Thank you very much. ahaha how did I forget to change `500 * time.Second` to `500 * time.Millisecond`??

The changes are quite comprehensive and nice, cool!

**Pull request merged**